tests/integration/targets/clickhouse_role/tasks/initial.yml

```
@@ -0,0 +1,54 @@
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Test
+- name: Create role in check mode
+  register: result
+  check_mode: true
+  community.clickhouse.clickhouse_role:
+    state: present
+    name: test_role
+
```

```suggestion
- name: Check return values
  ansible.builtin.assert:
    that:
    - result is changed
    - result.executed_statements == ['CREATE ROLE test_role']
```
in check_mode it shows the real work but doesn't change anything and we'll check it in your next test where we're querying the roles system table if it contains info about this role (i.e. if it was actually added or not and it shouldn't have been)